src/interp.cc

```diff
@@ -1337,6 +1337,8 @@ Result Thread::Run(int num_instructions) {
           } else {
             f = jit::compile(this, offset);
             env_->jit_compiled_functions_.insert({offset, f});
+            if (env_->trap_on_failed_comp && f == nullptr)
+              TRAP(FailedJITCompilation);
```

Comment:
This should be done using the `TRAP_IF` macro.

Reply:
Good point!

Reply:
Fixed.